R/add_coverage.R

```diff
@@ -58,7 +58,10 @@
 #' @export
 get_coverage <- function(data, by = "model") {
   # input checks ---------------------------------------------------------------
-  data <- as_forecast(na.omit(data), forecast_type = "quantile")
+  data <- copy(data)
+  data <- na.omit(data)
+  suppressWarnings(suppressMessages(validate_forecast(data)))
```

Comment:
Do we want to dispatch to the `quantile` validator here as a check that its the right kind of forecast type?

Reply:
Or happy to do the check you have below (though it would be nicer if it was in the validation step right?

Reply:
We could e.g. replace 
```
  data <- copy(data)
  data <- na.omit(data)
  suppressWarnings(suppressMessages(validate_forecast(data)))
  assert_subset(get_forecast_type(data), "quantile")
```
by 

```
  suppressWarnings(suppressMessages(as_forecast(data, forecast_type = "quantile")))
  data <- na.omit(data)
```
since `as_forecast()` automatically makes a copy. 
And then circle back once we have more clarity on #688 ?